src/Psl/Xml/Issue/IssueCollection.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Psl\Xml\Issue;
+
+use Countable;
+use IteratorAggregate;
+use Psl\Arr;
+use Psl\Iter;
+use Psl\Iter\Iterator;
+use Psl\Str;
+
+final class IssueCollection implements Countable, IteratorAggregate
+{
+    /**
+     * @var list<Issue>
+     */
+    private array $issues;
+
+    public function __construct(Issue ...$errors)
+    {
+        $this->issues = $errors;
+    }
+
+    /**
+     * @return Iterator<int, Issue>
+     */
+    public function getIterator(): Iterator
+    {
+        return Iterator::create($this->issues);
+    }
+
+    public function count(): int
+    {
+        return count($this->issues);
+    }
+
+    /**
+     * @param (callable(Issue): bool) $filter
+     */
+    public function filter(callable $filter): self
```

Comment:
Decided to go with a regular filter callback to give maximum flexibility

Reply:
even better :D